x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/components/phases/shared_fields/searchable_snapshot_field/searchable_snapshot_field.tsx

```diff
@@ -0,0 +1,231 @@
+          color="warning"
+          title={
+            <FormattedMessage
+              id="xpack.indexLifecycleMgmt.editPolicy.noSnapshotRepositoriesFoundTitle"
+              defaultMessage="No snapshot repositories found"
+            />
+          }
+        >
+          <FormattedMessage
+            id="xpack.indexLifecycleMgmt.editPolicy.noSnapshotRepositoriesFoundMessage"
+            defaultMessage="{link} to use searchable snapshots."
+            values={{
+              link: (
+                <EuiLink
+                  href={getUrlForApp('management', {
+                    path: `data/snapshot_restore/add_repository`,
+                  })}
+                  target="_blank"
+                >
+                  {i18n.translate(
+                    'xpack.indexLifecycleMgmt.editPolicy.createSearchableSnapshotLink',
+                    {
+                      defaultMessage: 'Create a snapshot repository',
+                    }
+                  )}
+                </EuiLink>
+              ),
+            }}
+          />
+        </EuiCallOut>
+      );
+    }
+  }
+
+  return (
+    <DescribedFormField
+      data-test-subj={`searchableSnapshotField-${phase}`}
+      switchProps={{
+        'data-test-subj': 'searchableSnapshotToggle',
+        label: i18n.translate(
+          'xpack.indexLifecycleMgmt.editPolicy.searchableSnapshotsToggleLabel',
+          { defaultMessage: 'Use searchable snapshot' }
+        ),
+        initialValue: Boolean(
+          policy.phases[phase]?.actions?.searchable_snapshot?.snapshot_repository
+        ),
+      }}
+      title={
+        <h3>
+          {i18n.translate('xpack.indexLifecycleMgmt.editPolicy.searchableSnapshotFieldTitle', {
+            defaultMessage: 'Searchable snapshot',
+          })}
+        </h3>
+      }
+      description={
+        <EuiTextColor color="subdued">
+          {i18n.translate(
+            'xpack.indexLifecycleMgmt.editPolicy.searchableSnapshotFieldDescription',
+            {
+              defaultMessage: 'Take a snapshot of the index and mount it as a searchable snapshot.',
+            }
+          )}
+        </EuiTextColor>
+      }
+      fullWidth
+    >
+      <div className="ilmSearchableSnapshotField">
+        {phase === 'hot' && isUsingSearchableSnapshotInHotPhase && (
+          <>
+            <EuiCallOut
+              data-test-subj="searchableSnapshotFieldsDisabledCallout"
+              title={i18n.translate(
+                'xpack.indexLifecycleMgmt.editPolicy.searchableSnapshotCalloutTitle',
+                { defaultMessage: 'Some actions have been disabled' }
+              )}
+              iconType="questionInCircle"
+            >
+              {i18n.translate('xpack.indexLifecycleMgmt.editPolicy.searchableSnapshotCalloutBody', {
+                defaultMessage:
+                  'Force merge, shrink, freeze and searchable snapshots are not allowed when this action is enabled in the hot phase.',
```

Comment:
```suggestion
                  'Force merge, shrink, and freeze are not allowed when searchable snapshots are enabled in the hot phase.',
```

Reply:
The toggle itself indicates *Use searchable snapshots*, which I believe is "this action" in the description. If that's the case, then I think we can remove the first mention of searchable snapshots and replace "this action" with "searchable snapshots".

Reply:
Thanks for the review here :). We still need to communicate that the searchable snapshot action in cold will also be disabled/disallowed - which what the origin "searchable_snapshot" text was trying to communicate 😄

Reply:
Will we use the same message for both hot/cold phase?

Will the message for searchable snapshots in the cold phase be conditional on the setting in the hot phase? For example, if searchable snapshots are disabled in the hot phase and users are enabling this setting only in the cold phase, it seems odd to show them a message about enabling searchable snapshots in the hot phase.

Similarly, if enabled in the hot phase, we should show a different message in the cold phase indicating why searchable snapshots are disabled.

### Enabled in hot phase, disabled in cold
**Hot phase**
[ x ] Create searchable snapshot
>Force merge, shrink, and freeze are not allowed when searchable snapshots are enabled in the hot phase. Also, searchable snapshots will be disabled in the cold phase.

**Cold phase**
[   ] Create searchable snapshot
>Searchable snapshots are disabled in the cold phase when already enabled in the hot phase.

### Disabled in hot phase, enabled in cold
**Hot phase**
[   ] Create searchable snapshot

**Cold phase**
[ x ] Create searchable snapshot

Reply:
> Will we use the same message for both hot/cold phase?

This particular message only applies to setting searchable snapshot in the hot phase.  Happy to add the additional message per your recommendation.

Reply:
I think it depends on how the UI responds. When the user enables searchable snapshots in the hot phase, will the ability to enable them in the cold phase be greyed out or made unavailable (not displayed)? If the former, I think it might be helpful for users to know why that option is greyed out. We mention in the searchable snapshots description in the hot phase that snapshots will be disabled in the cold phase, but users could miss that mention. 

I could go either way on this one, but I'm trying to think through the UX from the user's standpoint. If we remove the option for enabling searchable snapshots in the cold tier when that option is enabled in the hot tier, then users wouldn't have to question the option.

Reply:
Yeah totally, I agree. I have updated the PR description with some new screenshots for the "disabled" version of the control in cold phase (it is a callout instead of just a disabled control).